src/get_imports.rs

```diff
@@ -0,0 +1,244 @@
+use lazy_static::__Deref;
+use std::collections::{HashMap, HashSet};
+use std::iter::FromIterator;
+use std::panic;
+use std::path::{Path, PathBuf};
+use std::rc::Rc;
+use std::sync::Arc;
+use swc_common::errors::{ColorConfig, Handler};
+use swc_common::source_map::Pos;
+use swc_common::SourceFile;
+use swc_ecma_parser::Capturing;
+use swc_ecma_parser::{lexer::Lexer, Parser, StringInput, Syntax};
+
+use crate::error::{GetImportError, GetImportErrorKind};
+
+pub fn get_imports_map_from_file<'a>(
+    file_path: &'a PathBuf,
+) -> Result<HashMap<String, Option<HashSet<String>>>, GetImportError> {
+    get_imports_from_file(&file_path)
+}
+
+fn get_imports_from_file<'a>(
+    file_path: &'a PathBuf,
+) -> Result<HashMap<String, Option<HashSet<String>>>, GetImportError> {
+    let path_string = match file_path.to_str() {
+        Some(path) => path,
+        None => {
+            return Err(GetImportError::new(
+                GetImportErrorKind::ReadImportError,
+                None,
+                None,
+                None,
+            ))
+        }
+    };
+    let cm = Arc::<swc_common::SourceMap>::default();
+    let fm = match cm.load_file(Path::new(path_string)) {
+        Ok(f) => f,
+        Err(e) => {
+            return Err(GetImportError::new(
+                GetImportErrorKind::ReadTsFileError,
+                Some(path_string.to_string()),
+                None,
+                Some(vec![e]),
+            ));
+        }
+    };
+
+    let mut parser_errors: Vec<String> = Vec::new();
+
+    let dest_vector: Vec<u8> = Vec::new();
+
+    let dst = Box::new(dest_vector);
+
+    let handler = Handler::with_emitter_writer(dst, Some(cm.clone()));
```

Comment:
Why do we need to clone here?

Reply:
I'm not sure why, this was extracted from the swc docs.